components/zeekopenbsm/src/openbsmconsumer.cpp

```diff
@@ -0,0 +1,363 @@
+      event.type = Event::Type::Connect;
+    } else if (tok.tt.hdr64_ex.e_type == AUE_BIND) {
+      event.type = Event::Type::Bind;
+    }
+    break;
+  }
+}
+
+void OpenbsmConsumer::extractSubject(IOpenbsmConsumer::Event &event,
+                                     tokenstr_t tok) {
+
+  switch (tok.id) {
+  case AUT_SUBJECT32: {
+    uint32_t pid = tok.tt.subj32.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj32.euid;
+    event.header.group_id = tok.tt.subj32.egid;
+    break;
+  }
+  case AUT_SUBJECT64: {
+    uint32_t pid = tok.tt.subj64.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj64.euid;
+    event.header.group_id = tok.tt.subj64.egid;
+    break;
+  }
+  case AUT_SUBJECT32_EX: {
+    uint32_t pid = tok.tt.subj32_ex.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj32_ex.euid;
+    event.header.group_id = tok.tt.subj32_ex.egid;
+    break;
+  }
+  case AUT_SUBJECT64_EX: {
+    uint32_t pid = tok.tt.subj64_ex.pid;
+    event.header.process_id = pid;
+    event.header.path = getPathFromPid(pid);
+    event.header.user_id = tok.tt.subj64_ex.euid;
+    event.header.group_id = tok.tt.subj64_ex.egid;
+    break;
+  }
+  }
+}
+
+void OpenbsmConsumer::extractReturn(IOpenbsmConsumer::Event &event,
+                                    tokenstr_t tok) {
+
+  switch (tok.id) {
+  case AUT_RETURN32: {
+    int error = 0;
+    if (au_bsm_to_errno(tok.tt.ret32.status, &error) == 0) {
+      if (error == 0) {
+        event.header.success = 1;
+      } else {
+        event.header.success = 0;
+      }
+    } else
+      event.header.success = 0;
+    break;
+  }
+  case AUT_RETURN64: {
+    int error = 0;
+    if (au_bsm_to_errno(tok.tt.ret64.err, &error) == 0) {
+      if (error == 0) {
+        event.header.success = 1;
+      } else {
+        event.header.success = 0;
+      }
+    } else
+      event.header.success = 0;
+    break;
+  }
+  }
+}
+void OpenbsmConsumer::extractSocketInet(IOpenbsmConsumer::Event &event,
+                                        tokenstr_t tok) {
+
+  switch (tok.id) {
+  case AUT_SOCKINET32: {
+    if (event.type == Event::Type::Bind) {
+      event.header.remote_address = "0.0.0.0";
+      event.header.remote_port = 0;
+      event.header.local_address = getIpFromToken(tok);
+      event.header.local_port = ntohs(tok.tt.sockinet_ex32.port);
+    } else {
+      event.header.remote_address = getIpFromToken(tok);
+      event.header.remote_port = ntohs(tok.tt.sockinet_ex32.port);
+      event.header.local_address = "0.0.0.0";
+      event.header.local_port = 0;
+    }
+    if (tok.tt.sockinet_ex32.family == 2) {
+      event.header.family = 2;
+    } else if (tok.tt.sockinet_ex32.family == 26) {
+      event.header.family = 10;
+    } else {
+      event.header.family = 0;
+    }
```

Comment:
This is slightly hard to follow. Can you use some existing constants for the RHS?